### PR TITLE
UI Changes and Rotate Endpoint

### PR DESCRIPTION
**Pull request by Colin-Schultz**

### CODE REVIEW

File: fidelius-service/src/main/java/org/finra/fidelius/services/CredentialsService.java [hunk 1]
```diff
@@ -346,6 +353,39 @@ public Credential createCredential(Credential credential) {
...
+     * @param sourceType  Source type of the credential
+     * @param source      Source of the credential
+     * @param shortKey    Short key or name associated with credential
+     * @param component   Optional component associated with credential
+     * @param region      AWS Region associated with AWS Account
+     * @param application Key representing membership section of key
+     * @param environment Key representing the environment in which the credential is associated with
+     * @return Secret
+     */
+    //@PreAuthorize("@fideliusRoleService.isAuthorized(#application, #account)")
```
Comment (Brijeshrpatel9) [inline fidelius-service/src/main/java/org/finra/fidelius/services/CredentialsService.java:369]: Do we need this? How is authorisation handled for rotate?

Comment (Colin-Schultz) [reply fidelius-service/src/main/java/org/finra/fidelius/services/CredentialsService.java:369]: Ah we do, that was disabled for local testing. I'll fix that

File: fidelius-ui/src/app/edit/edit.component.html [hunk 2]
```diff
@@ -108,7 +158,7 @@
                              layout="column"
                              #secretRadioElement
                              name="secretTypeRadio"
-                             required
+                             required=false
```
Comment (Brijeshrpatel9) [inline fidelius-ui/src/app/edit/edit.component.html:161]: Not sure why are we making secretType as false? As of now, we have this as required.

File: fidelius-ui/src/mock-api/config.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
 	"note": "Mock API for Fidelius",
-	"mockDirectory": "",
+	"mockDirectory": "C:\\Users\\k29815\\github\\Fidelius\\fidelius-ui/src/mock-api/data/",
```
Comment (Brijeshrpatel9) [inline fidelius-ui/src/mock-api/config.json:3]: I would suggest to remove this.

File: fidelius-ui/src/app/edit/edit.component.html (hunk 2 above)
Comment (Colin-Schultz) [reply fidelius-ui/src/app/edit/edit.component.html:161]: This needs to be false to allow edits to just the metadata. There was additional logic added to send edits just for secrets, metadata, or both

File: fidelius-ui/src/mock-api/config.json (hunk 3 above)
Comment (Colin-Schultz) [reply fidelius-ui/src/mock-api/config.json:3]: Removed

**Pull request merged**